include/drivers/kinetis/ftm.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+#include <MK64F12.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+/*********
+ * TYPES
+ *********/
+typedef struct {
+    uint8_t ftm_num;                /* ftm num to configure */
+    uint32_t *ftm_clock_gate_base;
+    uint32_t ftm_clock_gate_mask;
+    uint32_t freq;                  /* Desired FTM Counter Frequency */
+    uint8_t prescalar;              /* 0-7 Divide by 2^PS */
+    uint8_t clksrc;                 /* 0-3 No CLK, SysCLK, FFCLK, EXCLK */
+    uint16_t cnt_init;              /* uint16 to initialize cnt */
+} ftm_config_t;
```

Comment:
This could be made smaller (in terms of memory) because there are several sub-byte fields which can be condensed - unfortunately that requires packing, and this is the clearer code.  I think we should stick with this way for now, but it might be worth considering the other way in the future.